src/dev-utils/deltareplayer.js

```diff
@@ -99,24 +104,28 @@ export default class DeltaReplayer {
 	 */
 	applyAllDeltas() {
 		return this.applyNextDelta()
-			.then( () => this.applyAllDeltas() )
-			.catch( () => {} );
+			.then( ( isFinished ) => {
+				if ( !isFinished ) {
+					return this.applyAllDeltas();
+				}
+			} );
 	}
 
 	/**
 	 * Applies the next delta to replay.
+	 * Returns promise with a `isFinished` parameter.
 	 *
-	 * @returns {Promise}
+	 * @returns {Promise.<Boolean>}
 	 */
 	applyNextDelta() {
 		const document = this._document;
 
```

Comment:
I think it's a safer way to do it. If we would want to change this class in the future and add some other `enqueuChanges()` modifying `this._deltasToReplay` then some strange errors could occur. (As far as I understand the idea behind `enqueueChanges` ;)